### PR TITLE
Move libtorch CMakeLists.txt to torch/

### PR DESCRIPTION
**Pull request by orionr**
Seeing if this quick move doesn't break anything. We'll be unifying this more with the root-level cmake file in the future.

cc @goldsborough @soumith

### CODE REVIEW

**Comment by goldsborough** [comment]
It seems generally fine but can I ask what the value is of doing this separate from the PR that does the actual unification? It will just be disturb my build scripts and I'll have to do fresh builds everywhere since the paths won't match anymore. It's a minor discomfort but just wondering why not do it all in one push

**Comment by orionr** [comment]
@goldsborough, I can see if I can call from root CMake  now, but full unification (adding Python builds here so it can replace _C) will likely take a bit more. I'm trying to break things up so PRs aren't so long-lived, but I do care about your discomfort.

Let me see if I can do the root CMake change - I'll likely need to add `BUILD_TORCH` as a flag for now, since we won't do this for setup.py right now.

**Comment by goldsborough** [comment]
Ok, let me know if this is inconvenient in any way, then it's no problem to merge this. Progress has precedence

**Comment by orionr** [comment]
After getting into it more, using root CMake is going to take some time. @soumith or @ezyang, can we do this to start? I'll continue working on the next step.

**Comment by soumith** [comment]
@orionr sure go ahead

File: torch/CMakeLists.txt [hunk 1]
```diff
@@ -55,7 +51,6 @@ if(USE_CUDA)
     ${TORCH_SRC_DIR}/../cmake/public
     ${TORCH_SRC_DIR}/../cmake/Modules_CUDA_fix
     /usr/lib/x86_64-linux-gnu/
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/TH/cmake
```
[the remark this answers is not in the document]
Comment (orionr) [reply torch/CMakeLists.txt]: It didn't seem like that location existed and definitely doesn't when I move it. @goldsborough can you confirm?

[the remark this answers is not in the document]
Comment (goldsborough) [reply torch/CMakeLists.txt]: yeah that seems like it was never used

**Review by soumith — APPROVED** [review]

**Pull request merged**